File: src/main/java/com/intellij/tapestry/intellij/view/nodes/MixinsNode.java
package com.intellij.tapestry.intellij.view.nodes;

import com.intellij.ide.projectView.PresentationData;
import com.intellij.openapi.module.Module;
import com.intellij.psi.PsiDirectory;
import com.intellij.tapestry.core.model.TapestryLibrary;
import icons.TapestryIcons;

public class MixinsNode extends PackageNode {

    public MixinsNode(TapestryLibrary library, PsiDirectory psiDirectory, Module module) {
        super(library, psiDirectory, module);

        init(psiDirectory, new PresentationData(psiDirectory.getName(), psiDirectory.getName(), TapestryIcons.Mixins, null));
    }

    public MixinsNode(PsiDirectory psiDirectory, Module module) {
        super(psiDirectory, module);

        init(psiDirectory, new PresentationData(psiDirectory.getName(), psiDirectory.getName(), TapestryIcons.Mixins, null));
    }
}

File: src/main/kotlin/com/github/rar91279/plugin/tapestry/intellij/view/nodes/MixinsNode.kt
package com.github.rar91279.plugin.tapestry.intellij.view.nodes

import com.github.rar91279.plugin.tapestry.core.model.presentation.PresentationLibraryElement
import com.intellij.openapi.module.Module
import com.intellij.psi.PsiDirectory
import com.intellij.ui.treeStructure.SimpleNode
import icons.TapestryIcons

/** The mixins of a module. */
class MixinsNode(
    module: Module,
    directories: List<PsiDirectory>,
    elements: Collection<PresentationLibraryElement>,
    showElementFiles: Boolean = true
) : ElementCategoryNode("Mixins", TapestryIcons.Mixins, module, directories, elements, showElementFiles) {

    override fun elementNode(element: PresentationLibraryElement, module: Module): SimpleNode =
        MixinNode(element, module, showElementFiles)
}
